### PR TITLE
DS: migrate the i18N settings page

### PR DESCRIPTION
**Pull request by mfrachet**
### What does it do?

Add the I18N list view

### Why is it needed?

You know why.

### How to test it?

Start the app and navigate to the settings section, then press the "internationalization" link

### Recording

Navigating the I18N List view setting using the keyboard and VoiceOver

### NOTE

Tests will be fixed when adding the modals to the view

### CODE REVIEW

File: packages/core/helper-plugin/lib/src/hooks/useFocusWhenNavigate/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+
+const useFocusWhenNavigate = (selector = 'main', dependencies = []) => {
```
Comment (mfrachet) [inline packages/core/helper-plugin/lib/src/hooks/useFocusWhenNavigate/index.js:3]: ℹ️ allows to focus the given selector (main in this scenario) content of the page when navigating using the keyboard

**Review by ronronscelestes — APPROVED** [review]
LGTM 🔥

File: packages/core/helper-plugin/lib/src/hooks/useFocusWhenNavigate/index.js (hunk 1 above)
Comment (HichamELBSI) [reply packages/core/helper-plugin/lib/src/hooks/useFocusWhenNavigate/index.js:3]: Very nice one. However, I'm not sure to understand in which case we need it. Isn't the role of the `skipToContent` button ?

File: packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                  <ActionIconWrapper reverse={isReversedArrow(SortingKeys.name)}>
+                    <DropdownIcon />
+                  </ActionIconWrapper>
+                }
+                noBorder
+                onClick={() => handleSorting(SortingKeys.name)}
+              />
+            }
+          >
+            <TableLabel textColor="neutral600">Display name </TableLabel>
```
Comment (HichamELBSI) [inline packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js:93]: Same

File: packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                  <ActionIconWrapper reverse={isReversedArrow(SortingKeys.default)}>
+                    <DropdownIcon />
+                  </ActionIconWrapper>
+                }
+                noBorder
+                onClick={() => handleSorting(SortingKeys.default)}
+              />
+            }
+          >
+            <TableLabel textColor="neutral600">Default</TableLabel>
```
Comment (HichamELBSI) [inline packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js:109]: Same about translated messages

File: packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+          </Th>
+          <Th>
+            <VisuallyHidden>Actions</VisuallyHidden>
```
Comment (HichamELBSI) [inline packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js:112]: Same

File: packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                  <ActionIconWrapper reverse={isReversedArrow(SortingKeys.id)}>
+                    <DropdownIcon />
+                  </ActionIconWrapper>
+                }
+                noBorder
+                onClick={() => handleSorting(SortingKeys.id)}
+              />
+            }
+          >
+            <TableLabel textColor="neutral600">ID</TableLabel>
```
Comment (HichamELBSI) [inline packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js:77]: Not sure about the ID TBH but I think we should use translated message here

File: packages/plugins/i18n/admin/src/components/LocaleList/index.js [hunk 6]
```diff
@@ -1,89 +1,85 @@
...
+        primaryAction={
+          <Button startIcon={<AddIcon />} onClick={onToggleCreateModal}>
+            {formatMessage({ id: getTrad('Settings.list.actions.add') })}
+          </Button>
+        }
+        title={formatMessage({ id: getTrad('plugin.name') })}
+        subtitle={formatMessage({ id: getTrad('Settings.list.description') })}
       />
+      <ContentLayout>
+        {locales?.length === 0 ? (
```
Comment (HichamELBSI) [inline packages/plugins/i18n/admin/src/components/LocaleList/index.js:44]: Can't we runonly one  condition ? 
```js
{locales?.length === 0 ? <EmptyState /> : <Table />}
```

File: packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+          </Th>
```
Comment (bulby97) [inline packages/plugins/i18n/admin/src/components/LocaleList/LocaleTable.js:78]: It might be a good idea to extract this component at some point

**Review by bulby97 — APPROVED** [review]
LGTM :)

**Review by HichamELBSI — CHANGES_REQUESTED** [review]
Very nice! Good job @mfrachet. I just put small comments about translated messages but it looks pretty good to me. 
Also, I have a feedback about the visual focus indicator on the table header. It seems really big to me WDYT?

Screenshot 2021-08-16 at 14 29 18

**Comment by mfrachet** [comment]
Concerning the icon button close to the TH, I think it's caused by something inside core, like a CSS from bootstrap or something like that. In the DS, it looks okay, to me. WYT? https://design-system-git-develop-strapijs.vercel.app/?path=/story/design-system-organisms-table--with-th-actions

**Comment by HichamELBSI** [comment]
> Concerning the icon button close to the TH, I think it's caused by something inside core, like a CSS from bootstrap or something like that. In the DS, it looks okay, to me. WYT? https://design-system-git-develop-strapijs.vercel.app/?path=/story/design-system-organisms-table--with-th-actions

Indeed 😅

**Review by HichamELBSI — APPROVED** [review]
Good job 👍🏽

**Pull request merged**